### PR TITLE
doc: USB state power manager documentation

### PR DESCRIPTION
**Pull request by rakons**
Add documentation for the USB state power manager module.

JIRA: NCSDK-10557

Signed-off-by: Radoslaw Koppel <radoslaw.koppel@nordicsemi.no>

### CODE REVIEW

File: applications/nrf_desktop/doc/usb_state_pm.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+The |usb_state_pm| is small, stateless module, that manages power state required by
+the USB state module.
+
+Configuration
+*************
+
+The module is enabled by selecting :kconfig:`CONFIG_DESKTOP_USB_PM_ENABLE`.
+It depends on :kconfig:`CONFIG_DESKTOP_USB_ENABLE` and :kconfig:`CONFIG_CAF_POWER_MANAGER`.
+
+There is no other configuration.
```
Comment (pdunaj) [inline applications/nrf_desktop/doc/usb_state_pm.rst:19]: I am not sure these two lines are needed.

**Review by pdunaj — APPROVED** [review]

File: applications/nrf_desktop/doc/usb_state_pm.rst (hunk 1 above)
Comment (rakons) [reply applications/nrf_desktop/doc/usb_state_pm.rst:19]: I would leave the line that describes that logs are inherited from USB state module. If anybody would look for that setting. The information that there is no other configuration may be removed.

**Review by greg-fer — APPROVED** [review]
Approving, but please check my edits and squash the changes. The changelog commit assumes the Bluetooth state module PR will go in first and that this PR goes in after 1.7.0, so I'm adding also the DNM.

**Comment by rakons** [comment]
@greg-fer Didn't you mean to put this comment into #5550? This is USB documentation.

**Comment by greg-fer** [comment]
> Didn't you mean to put this comment into #5550? 

You mean about the approval and changelog commit? I've added the empty changelog in this PR with entries from #5550 already added, though without links. Either can go in first, as long as they go more or less together. We can revert if you feel this is wrong.

**Comment by rakons** [comment]
Ok - waiting for release anyway - I will rebase it then as release-notes-changelog.rst  would be cleared anyway.

**Comment by greg-fer** [comment]
Rebased to the updated, emptied changelog.

**Comment by greg-fer** [comment]
Removing the DNM label I put before the release.

**Comment by rakons** [comment]
Documentation recheck request - only changelog, the previous one was broken (during rebase maybe?)

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -54,9 +54,14 @@ nRF5
 nRF Desktop
 -----------
 
+* Added:
+
+  * Added missed documentation for :ref:`nrf_desktop_usb_state_pm`.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:59]:
```suggestion
  * Added documentation for :ref:`nrf_desktop_usb_state_pm`.
```

**Review by greg-fer — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**